ukoly/HanaPospisilova/ukol-4/ukol.md

```diff
@@ -0,0 +1,25 @@
+- na serveru články obsahují id, titulek, autor, datum, perex, text, kategorii, imgUrl, boolean pro hlavní článek (každá kategorie má jeden hlavní článek)
+
+1. Třída Post (pro vytvoření jednoho článku)
+   - do konstruktoru nadpis, autor, datum, perex a text
+
+2. Třída RandomPost - podtřída třídy Post
+   - do konstruktoru navíc kategorie a imgUrl
+
+3. 3 ruzné templaty
+   - hlavní článek - používá instanci Post
+   - pro článek ve vybraných dvou - používá instanci RandomPost
+   - pro normální článek ve výpisu - používá instanci Post
+
+4. Třída App
+   - do konstruktoru dostane DOM elementy sekcí, url na server,
+   - metoda run vyrenderuje pomocí dalších metod hlavní článek, vybrané 2 články a další články a taky menu
+   - další metody: showPosts - zobrazí výpis článků podle zadaného endpointu
+     showMainPost - zobrazí hlavní článek (dotaz na server na příslušný boolean atribut)
+     showRandomPosts - 2 náhodně vybrané články, každý z jiné kategorie
+     metoda na vytvoření event listeneru na filtrování po kliku na položku menu
+
+5. Třída Menu
```

Comment:
Doporučuji vytvořit kategorie jako samostatný model stejně jako jsou články.

Např. struktura bude `id`, `name`... takže třeba `id:1`, `name:World`

Pak se ale musí vyřešit párování článek x kategorie. Zde je několik způsobů.
- atribut kategorie bude ID

```json
{
  "id": 1,
  "title": "Můj první článek",
  "category": 2
}
```

- celá struktura kategorie

```json
{
  "id": 1,
  "title": "Můj první článe",
  "category": {
    "id": 2,
    "name": "World"
  }
}
```

- klíčové slovo, které nám bude pomáhat vyhledat články v kategorii

```json
{
  "id": 1,
  "title": "Můj první článek",
  "category": "world"
}
```

Uložení kategorie bych směřoval k tomu, jak budeš podle něj vyhledávat.